src/backend/commands/trigger.c

```diff
@@ -5575,7 +5575,7 @@ AfterTriggerFreeQuery(AfterTriggersQueryData *qs, bool free_tables)
 		tuplestore_end(ts);
 
 	/* Babelfish triggers will remove transition tables later */
-	if (!free_tables)
+	if (!free_tables && afterTriggers.query_depth >= compositeTriggers.triggers->query_depth)
```

Comment:
it should == not >=. Also, we should write a new function like FreeTriggerTables which will check for this condition and iscompositeTrigger active